### PR TITLE
add controller

### PR DESCRIPTION
**Pull request by aidandyde**

### CODE REVIEW

File: templates/ad-lotto-theme/index.html.twig [hunk 1]
```diff
@@ -0,0 +1,172 @@
+<!DOCTYPE html>
```
Comment (apatkinson) [inline templates/ad-lotto-theme/index.html.twig]: add in `{% block title %}add title here{% endblock %}`

**Comment by mikekirchem** [comment]
add an ``.editorconfig`` file and your editor should start doing some helpful stuff automatically for you - https://editorconfig.org

File: templates/ad-lotto-theme/register.html.twig [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+</head>
+<body>
+	<!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
+    <!-- Include all compiled plugins (below), or include individual files as needed -->
+    <script src="js/bootstrap.min.js"></script>
+    <div class="col-lg-4 col-md-6">
+        <h2>Sign Up</h2>
+        <p>Please fill this form to create an account.</p>
+		<form action="<?php echo htmlspecialchars($_SERVER["PHP_SELF"]); ?>" method="post">
```
Comment (mikekirchem) [inline templates/ad-lotto-theme/register.html.twig]: ``<?php`` not valid in twig ?

File: templates/ad-lotto-theme/index.html.twig [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      <div class="container">
+        <a class="navbar-brand js-scroll-trigger" href="#page-top">AdLotto</a>
+        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
+          <span class="navbar-toggler-icon"></span>
+        </button>
+        <div class="collapse navbar-collapse" id="navbarResponsive">
+          <ul class="navbar-nav ml-auto">
+            <li class="nav-item">
+              <a class="nav-link js-scroll-trigger" href="#about">About</a>
+            </li>
```
Comment (sjdawson) [inline templates/ad-lotto-theme/index.html.twig]: Twig is beautiful in that it allows you to minimise the amount of repetition you have in a template. It probably wouldn't be a huge amount of a benefit here, but this markup can show how it would be handy. For example, at the top of this file, you could have the following block:

```twig
{% macro navitem(href, text) %}
<li class="nav-item">
    <a class="nav-link js-scroll-trigger" href="{{ href }}">{{ text }}</a>
</li>
{% endmacro %}
{% import _self as partials %}
```

Then, this section would become:

```twig
<ul class="navbar-nav ml-auto">
    {{ partials.navitem('#about', 'About') }}
    {{ partials.navitem('#services', 'Prizes') }}
    {{ partials.navitem(path('register'), 'Sign Up') }}
    {{ partials.navitem(path('login'), 'Log In') }}
</ul>
```

Now, if you ever wanted to change the markup of your navigation items, for example, adding a new class to the `a` element, or wrapping the text into a `span` element for further tweaking, you'd only need to do this once, instead of multiple times, risking missed elements or typos. The macro would be updated, and all of your navigation elements would follow suit instantly:

```twig
{% macro navitem(href, text) %}
<li class="nav-item my-new-class-for-styling">
    <a class="nav-link js-scroll-bounce" href="{{ href }}"><strong>{{ text }}</strong></a>
</li>
{% endmacro %}
```

(Added `my-new-class-for-styling` to the `li`, changed `js-scroll-trigger` to `js-scroll-bounce` and wrapped the text element in a `strong` tag)

It also makes adding a new element to your navigation trivially easy, too.

File: templates/ad-lotto-theme/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        </div>
+      </div>
+    </nav>
+
+    <header class="masthead text-center text-white d-flex">
+      <div class="container my-auto">
+        <div class="row">
+          <div class="col-lg-8 mx-auto">
+            <h1>
+              <strong class="strong-dark">Ad</strong><strong>Lotto</strong>
```
Comment (sjdawson) [inline templates/ad-lotto-theme/index.html.twig]: I'd have used the following markup for this element:
```twig
<!-- Can't remember if this is a legit style from bootstrap, if not, I'd add it to custom styling for reuse -->
<h1 class="text-strong">     
    <span class="text-dark">Ad</span>Lotto
</h1>
```
And then re-work your CSS classes to match. 
* The `strong` element in here isn't really needed to convey the added emphasis it does for screen-readers. It could be applied by the class `text-strong`:

```css
.text-strong { font-weight: 700 }
```
* `strong-dark` essentially ties that class to the element, where the class doesn't set any particular weight. This class would look wrong, almost, if placed onto a `p` tag, for instance - and it didn't set the font weight to bold.

File: templates/ad-lotto-theme/index.html.twig [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            <h2 class="section-heading">Win Big!</h2>
+            <hr class="my-4">
+          </div>
+        </div>
+      </div>
+      <div class="container">
+        <div class="row">
+          <div class="col-lg-4 col-md-8 text-center">
+            <div class="service-box mt-5 mx-auto">
+              <img src="{{ asset('build/images/calendar.png') }}" alt="" class="img-fluid" style='height: 100%; width: 100%; object-fit: contain; padding: 30px'>
```
Comment (sjdawson) [inline templates/ad-lotto-theme/index.html.twig]: Inline styling is a good ground to quickly test some styling, but ideally, you'll want to be moving those styles to CSS when you're done. Inline styles have the highest of specificity, so it's hard to override them with another stylistic choice. For example, if you set `padding: 50px` on the class `img-fluid` in your CSS, it would never be represented, since this inline style would override it.

File: templates/ad-lotto-theme/register.html.twig (hunk 2 above)
Comment (apatkinson) [reply templates/ad-lotto-theme/register.html.twig]: I think its left over from a big copy paste

File: src/Repository/UserRepository.php [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, User::class);
+    }
+
+//    /**
+//     * @return User[] Returns an array of User objects
+//     */
+    /*
+    public function findByExampleField($value)
```
Comment (Jamesking56) [inline src/Repository/UserRepository.php]: You can save some space by removing this old code

File: assets/css/creative.css [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+  -webkit-transition: all 0.2s;
+  -moz-transition: all 0.2s;
+  transition: all 0.2s;
+}
+
+a:hover {
+  color: #4bea50;
+}
+option,
+dropdown,
```
Comment (sjdawson) [inline assets/css/creative.css]: This one should never target anything, as `dropdown` isn't a valid HTML element. You might have been looking to use `select` here instead.

Comment (aidandyde) [reply assets/css/creative.css]: I think it was a bootstrap element or perhaps a class that got left in there from when I copied it over

File: src/Entity/User.php [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+        $this->registeredOn = new \DateTime();
+        $this->email = $email;
+        $this->username = $username;
+        $this->password = $password;
+        $this->referrer = $referrer;
+        $this->currency = $currency;
+        $this->entries = 0;
+
+        return $this;
```
Comment (cs278) [inline src/Entity/User.php]: No need for a return in a constructor.

File: src/Entity/User.php [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    public function setRegisteredOn(\DateTimeInterface $registeredOn): self
+    {
+        $this->registeredOn = $registeredOn;
+
+        return $this;
+    }
+
+    public function getReferrer(): ?interedThisWeek
```
Comment (cs278) [inline src/Entity/User.php]: auto complete typo

**Comment by apatkinson** [comment]
Check out:

https://symfony.com/doc/current/contributing/code/standards.html

Will help to keep everything neat!
